src/Microsoft.TemplateEngine.Abstractions/IGenerator.cs

```diff
@@ -55,8 +60,41 @@ public interface IGenerator : IIdentifiedComponent
         /// <param name="environmentSettings">template engine environment settings.</param>
         /// <param name="template">template to get parameters from.</param>
         /// <returns><see cref="IParameterSet"/> with parameters available in <paramref name="template"/>.</returns>
+        [Obsolete("Replaced by ParameterSetBuilder", false)]
         IParameterSet GetParametersForTemplate(IEngineEnvironmentSettings environmentSettings, ITemplate template);
 
+        /// <summary>
+        /// Generates the <paramref name="template"/> with given parameters.
+        /// </summary>
+        /// <param name="environmentSettings">template engine environment settings.</param>
+        /// <param name="template">template to generate.</param>
+        /// <param name="parameters">template parameters.</param>
+        /// <param name="targetDirectory">target output directory to generate template to.</param>
+        /// <param name="cancellationToken">cancellation token.</param>
+        /// <returns><see cref="ICreationResult"/> containing post actions and primary outputs after template generation.</returns>
+        Task<ICreationResult> CreateAsync(
```

Comment:
Generator needs to do the filtering of symbols from template (due to current renaming logic, that get's info from symbols, not parameters or variables - this would be good to rework as well, but I decided to scope this out as it is hodden detail). For that reason it needs evaluated data. Casing or evaluation within generator are viable alternatives